docs/BackendAPI.md

```diff
@@ -0,0 +1,52 @@
+# ONNX Backend Interface Proposal
+
+We propose a cross-platform API for loading and executing ONNX graphs on optimized backends. High-level frameworks and applications can use this API to execute neural network and machine learning models. Hardware vendors can implement this API to expose specialized hardware accelerators and highly optimized software infrastructure to the users.
+
+## Core Features
+
+- Standardized interface for neural network inference on special-purpose accelerators, CPUs, GPUs, DSPs, and FPGAs
+- Based on widely supported technologies
+  - C API for function calls
+  - ONNX format for passing model graphs
+  - NCHW tensor layout for passing inputs and outputs
+- Dynamic discovery of available backends for model execution
+  - Multiple backends from different vendors can co-exist on the same system
+- Dynamic discovery of supported ONNX Operators on each backend
+
+### Optional Features:
+
+- Graphs with variable-shape inputs and/or outputs
+- Graphs with data-dependendent output shapes
+
+## How to Use ONNX Backend Interface
```

Comment:
This looks good, but what about allocating/deallocation the actual input/output tensors?

Reply:
Memory is allocated by the caller always. Even for dynamic output sizes we propose the caller to allocate maximally possible buffer beforehand. It's a tradeoff but allows to cut off extra level of complexity with managing memory

Reply:
OK.